integration-test/src/main/java/com/sequenceiq/it/cloudbreak/ImageCatalogTests.java

```diff
@@ -0,0 +1,209 @@
+                .withUrl(INVALID_IMAGECATALOG_URL), "an imagecatalog request with url no protocol"
+        );
+        when(ImageCatalog.post(), "post the imagecatalog request");
+        then(ImageCatalog.assertThis(
+                (imageCatalog, t) -> {
+                    Assert.assertNotEquals(imageCatalog.getResponse().getName(), VALID_IMAGECATALOG_NAME + "-url-np");
+                }),  "check imagecatalog is not created with url no protocol");
+    }
+
+    @Test(expectedExceptions = BadRequestException.class)
+    public void testCreateDeleteValidCreateAgain() throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(ImageCatalog.isCreatedDeleted()
+                .withName(VALID_IMAGECATALOG_NAME + "-delete-create")
+                .withUrl(VALID_IMAGECATALOG_URL), "an imagecatalog request then delete"
+        );
+        given(ImageCatalog.request()
+                .withName(VALID_IMAGECATALOG_NAME + "-delete-create")
+                .withUrl(INVALID_IMAGECATALOG_URL), "same imagecatalog request again"
+        );
+        when(ImageCatalog.post(), "post the imagecatalog request");
+        then(ImageCatalog.assertThis(
+                (imageCatalog, t) -> {
+                    Assert.assertEquals(imageCatalog.getResponse().getName(), VALID_IMAGECATALOG_NAME + "-delete-create");
+                }),  "check imagecatalog is created when name was used and deleted before");
+    }
+
+    @Test(expectedExceptions = BadRequestException.class)
+    public void testDeleteCbDefaultImageCatalog() throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(ImageCatalog.request()
+                .withName(DEFAULT_IMAGECATALOG_NAME)
+        );
+        when(ImageCatalog.delete(), "try to delete cb default image catalog");
+    }
+
+    @Test
+    public void testSetNewDeleteDefaultImageCatalog() throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(ImageCatalog.request()
+                .withName(VALID_IMAGECATALOG_NAME + "default")
+                .withUrl(VALID_IMAGECATALOG_URL), "an imagecatalog request and set as default"
+
+        );
+        when(ImageCatalog.post());
+        when(ImageCatalog.setDefault());
+        then(ImageCatalog.assertThis(
+                (imageCatalog, t) -> {
+                    Assert.assertEquals(imageCatalog.getResponse().getName(), VALID_IMAGECATALOG_NAME + "default");
+                    Assert.assertEquals(imageCatalog.getResponse().isUsedAsDefault(), true);
+                    try {
+                        given(ImageCatalog.request().withName(VALID_IMAGECATALOG_NAME + "default"));
+                        when(ImageCatalog.delete());
+                    } catch (Exception e) {
+                        LOGGER.error("Error occured during delete: " + e.getMessage());
+                    }
+                }),  "check imagecatalog is created, set as default and deleted");
+    }
+
```

Comment:
What do you think about an invalid image catalog JSON test? where the json has some invalid content 
I don't know maybe:
- invalid or unaccessible image url (won't hurt if CB validate these... )
- invalid schema
- missing tags